src/components/sliders/draw.js

```diff
@@ -36,10 +36,20 @@ module.exports = function draw(gd) {
         .classed(constants.containerClassName, true)
         .style('cursor', 'ew-resize');
 
-    sliders.exit().remove();
+    function clearSlider(sliderOpts) {
+        if(sliderOpts._commandObserver) {
+            sliderOpts._commandObserver.remove();
+            delete sliderOpts._commandObserver;
+        }
+
+        Plots.autoMargin(gd, autoMarginId(sliderOpts));
```

Comment:
Could we add a comment similar to what's above https://github.com/plotly/plotly.js/pull/2681/files#r191879632 here?

Reply:
sure -> https://github.com/plotly/plotly.js/pull/2681/commits/230a782766ae9864eb5ac803c5fb069bedb9ab25